cmd/stree.go

```diff
@@ -0,0 +1,61 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/xlab/treeprint"
+
+	apitypes "github.com/puppetlabs/wash/api/types"
+	cmdutil "github.com/puppetlabs/wash/cmd/util"
+	"github.com/spf13/cobra"
+)
+
+func streeCommand() *cobra.Command {
+	streeCmd := &cobra.Command{
+		Use:   "stree [<path>...]",
+		Short: "Displays the entry's stree (schema-tree)",
+		Long: `Displays the entry's stree (schema-tree), which is a high-level overview of the entry's
+hierarchy. Non-singleton types are bracketed with "[]".`,
+		RunE: toRunE(streeMain),
+	}
+	return streeCmd
+}
+
+func streeMain(cmd *cobra.Command, args []string) exitCode {
+	paths := args
+	if len(paths) == 0 {
+		paths = []string{"."}
+	}
+	conn := cmdutil.NewClient()
+	for _, path := range paths {
+		schema, err := conn.Schema(path)
+		if err != nil {
+			cmdutil.ErrPrintf("%v\n", err)
+			return exitCode{1}
+		}
+		stree := treeprint.New()
+		fill(stree, schema)
+		stree.SetValue(path)
+		cmdutil.Print(stree.String())
+	}
+	return exitCode{0}
+}
+
+func fill(stree treeprint.Tree, schema apitypes.EntrySchema) treeprint.Tree {
+	value := schema.Label
+	if len(value) == 0 {
```

Comment:
Think we should just require `Label` (i.e. panic if it is not provided). I see no good reason to default to printing Type.